src/main/java/de/undercouch/gradle/tasks/download/DownloadAction.java

```diff
@@ -168,7 +171,23 @@ private void execute(URL src, HttpClientFactory clientFactory) throws IOExceptio
                         + "progress will not be displayed.");
             }
         }
-        
+
+        if ("file".equals(src.getProtocol())) {
```

Comment:
I would prefer if we had a short if-else-block here and then call new methods `executeFileProtocol` and `executeHttpProtocol` maybe.